src/overlays/actors/ovl_Boss_Tw/z_boss_tw.c

```diff
@@ -156,183 +187,5337 @@ static ColliderCylinderInit D_8094A840 = {
+            }
+
+            Math_ApproachF(&this->subCamAt.y, sKoumePtr->actor.world.pos.y + 17.0f, 0.05f, 10.0f);
+
+            if (this->work[CS_TIMER_2] >= 50) {
+                Math_ApproachF(&this->subCamDist, 110.0f, 0.05f, this->subCamDistStep);
+                Math_ApproachF(&this->subCamDistStep, 1.0f, 1.0f, 0.025f);
+                this->subCamEye.x = this->subCamDist * sinf(this->subCamYaw);
+                this->subCamEye.z = this->subCamDist * cosf(this->subCamYaw);
+                if (this->work[CS_TIMER_2] >= 151) {
+                    this->subCamYaw += this->subCamYawStep;
+                    if (this->work[CS_TIMER_2] >= 800) {
+                        Math_ApproachF(&this->subCamYawStep, 0.0f, 1.0f, 0.0001f);
+                    } else {
+                        Math_ApproachF(&this->subCamYawStep, 0.015f, 1.0f, 0.0001f);
+                    }
+                }
+            }
+            Math_ApproachF(&sKoumePtr->actor.scale.x, 0.009999999f, 0.1f, 0.001f);
+            Actor_SetScale(&sKoumePtr->actor, sKoumePtr->actor.scale.x);
+            Actor_SetScale(&sKotakePtr->actor, sKoumePtr->actor.scale.x);
+            if (this->work[CS_TIMER_2] >= 1020) {
+                Camera* cam = Gameplay_GetCamera(globalCtx, 0);
+                cam->eye = this->subCamEye;
+                cam->eyeNext = this->subCamEye;
+                cam->at = this->subCamAt;
+                func_800C08AC(globalCtx, this->subCamId, 0);
+                this->csState2 = 4;
+                this->subCamId = 0;
+                func_80064534(globalCtx, &globalCtx->csCtx);
+                func_8002DF54(globalCtx, &this->actor, 7);
+                Audio_QueueSeqCmd(0x21);
+                Actor_SpawnAsChild(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_DOOR_WARP1, 600.0f, 230.0f,
+                                   0.0f, 0, 0, 0, -1);
+                Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_ITEM_B_HEART, -600.0f, 230.f, 0.0f, 0, 0, 0, 0);
+                this->actor.world.pos.y = -2000.0f;
+                this->workf[UNK_F18] = 0.0f;
+                sKoumePtr->visible = sKotakePtr->visible = false;
+                if (&this->subCamEye) {} // fixes regalloc, may be fake
+                Flags_SetClear(globalCtx, globalCtx->roomCtx.curRoom.num);
+            }
+            break;
+        case 4:
+            sEnvType = 0;
+            break;
+    }
+
+    if (this->subCamId) {
+        if (1) {}
+        Gameplay_CameraSetAtEye(globalCtx, this->subCamId, &this->subCamAt, &this->subCamEye);
+    }
+}
+#else
+Vec3f D_8094A8E8 = { 0.0f, 0.0f, 0.0f };
+Vec3f D_8094A8F4 = { 0.0f, 0.0f, 0.0f };
+#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Tw/BossTw_TwinrovaDeathCS.s")
+#endif
+
+static s16 D_8094A900[] = {
+    0x0000, 0x0001, 0x0002, 0x0002, 0x0001, 0x0000,
+};
+
+static s16 D_8094A90C[] = {
+    0x0000, 0x0001, 0x0002, 0x0002, 0x0002, 0x0002, 0x0002, 0x0002, 0x0001, 0x0000,
+};
```

Comment:
```suggestion
static s16 D_8094A900[] = { 0, 1, 2, 2, 1, 0};
static s16 D_8094A90C[] = {0, 1, 2, 2, 2, 2, 2, 2, 1, 0};
```
Those zeroes at the end might be padding. Are they ever indexed?